app/controllers/confirmation_controller.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class ConfirmationController < ApplicationController
+  rescue_from ActiveRecord::RecordNotUnique, with: :error_not_unique
+
+  # This populates instance variables used by the new confirmation form.
+  def new
+    @term = Term.find(params[:term_id])
+    @categories = Category.all
+    @confirmation = Confirmation.new
+  end
+
+  # This receives the submission from the new confirmation form, creating the needed record with the help of various
+  # private methods.
+  def create
+    confirmation = Confirmation.new({
+                                      term_id: params[:term_id],
+                                      user: current_user
+                                    })
+    set_category(confirmation, params[:confirmation][:category])
+    feedback_for(confirmation.save)
+    redirect_to terms_unconfirmed_path
+  end
+
+  private
+
+  # This method takes the result of the confirmation.save directive above and sets an appropriate flash message.
+  #
+  # The final else clause is likely to be difficult to provoke, but I have trouble building an if conditional without
```

Comment:
It's probably best to avoid first person when writing documentation.

See my comment about using Sentry. I think this documentation line could be removed if we log to Sentry when that state occurs as it should be clear to developers that it's not normal if Sentry is involved.